### ISSUE TITLE
Auto-detect blocked sites not always works correctly

### ISSUE THREAD

**Issue by Centaur** [getlantern/lantern#3028]
Sometimes Lantern recognize local sites as blocked and make the connection slow by trying to proxy it. As an example, try `yunpan.360.cn`. Lantern version: Lantern 2.0.1 (20150821.200810).
Or you may consider bring back the Lantern 1.x feature that let us user config what sites need to be proxied.

**Comment by Centaur** [comment]
Potential duplicate of #2999.  I always use Lantern in Firefox.

**Comment by myleshorton** [comment]
@fffw can you take a look at this?

**Comment by fffw** [comment]
It's definitely one of the reason to rewrite detour! By comparing the
actual response time, we can avoid proxy a slow site.

**Comment by skivvies** [comment]
just came across [cow ("Climb-Over-the-Wall proxy")](https://github.com/cyfdecyf/cow/blob/master/README-en.md) and thought it might be of interest, especially the
> It tries to automatically identify blocked websites and only use parent proxy for those sites.

in case there's anything there Lantern can use. It's written in Go too.

**Comment by myleshorton** [comment]
Nice thanks @skivvies! Very cool it includes Shadowsocks compatibility too.

**Comment by oxtoacart** [comment]
Blocked by #3296.
